test/unit/specs/components/staking/PageValidator.spec.js

```diff
@@ -277,209 +470,478 @@ describe(`onStake`, () => {
+            commit: jest.fn(),
+            dispatch: jest.fn(() => {
+              throw new Error(`one\ntwo\nthree\nfour\nfive\nsix\nseven`)
+            }),
+            getters: getterValues
+          }
 
-        expect($store.commit.mock.calls).toEqual([
-          [
-            `notify`,
-            {
-              body: `You have successfully staked your atoms.`,
-              title: `Successful Staking!`
+          const {
+            vm: { submitDelegation }
+          } = mount(PageValidator, {
+            mocks: {
+              $route: {
+                params: { validator: `1a2b3c`, delegator_shares: `19` }
+              },
+              $store
             }
-          ]
-        ])
+          })
+
+          let stakingTransactions = {}
+          stakingTransactions.delegations = [{ atoms: 10, validator: delegate }]
+
+          await submitDelegation({
+            amount: 10,
+            from: `cosmosaccaddr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctpesxxn9`
+          })
+
+          expect($store.dispatch.mock.calls).toEqual([
+            [`submitDelegation`, { stakingTransactions }]
+          ])
+
+          expect($store.commit.mock.calls).toEqual([
+            [
+              `notifyError`,
+              {
+                body: `six`,
+                title: `Error while delegating ${getterValues.bondingDenom}s`
+              }
+            ]
+          ])
+        })
       })
 
-      it(`error`, async () => {
-        const $store = {
-          commit: jest.fn(),
-          dispatch: jest.fn(() => {
-            throw new Error(`message`)
-          }),
-          getters: getterValues
-        }
+      describe(`composition`, () => {
+        it(`delegation.submitDelegation`, async () => {
+          const delegation = Delegation({})
 
-        const {
-          vm: { submitDelegation }
-        } = mount(PageValidator, {
-          mocks: {
-            $route: { params: { validator: `1a2b3c` } },
-            $store
+          const dispatch = jest.fn((type, payload) => {
+            if (type === `submitDelegation`) {
+              delegation.actions[type]($store, payload)
+            }
+          })
+
+          const $store = {
+            commit: jest.fn(),
+            dispatch,
+            getters: getterValues,
+            rootState: getterValues,
+            state: {
+              committedDelegates: { "1a2b3c": 0 },
+              unbondingDelegations: {}
+            }
           }
-        })
 
-        await submitDelegation({ amount: 10 })
+          const {
+            vm: { submitDelegation }
+          } = mount(PageValidator, {
+            mocks: {
+              $route: { params: { validator: `1a2b3c` } },
+              $store
+            }
+          })
+
+          await submitDelegation({
+            amount: 10,
+            from: `cosmosaccaddr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctpesxxn9`
+          })
+
+          expect($store.dispatch.mock.calls).toEqual([
```

Comment:
Showing whole json objects in the tests create a lot of noise imo. Can we change this to test only what is necessary? Like the testing, that the validator is correct by comparing the owner instead of posting the whole validator. Plus adding a snapshot test.

Reply:
This was already like this. No need to change in this PR.